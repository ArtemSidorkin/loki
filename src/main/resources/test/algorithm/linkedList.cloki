List:

	Node value next:
		$value = value
		$next = next

	$__root = none
	$__size = 0

	$size: $__size

	$add value:
		nwNd = Node.new(value, none)
		if ($__root == none) $__root = nwNd
		else
			itrtrNd = $__root
			while(itrtrNd.next != none) itrtrNd = itrtrNd.next
			itrtrNd.next = nwNd
		$__size = $__size + 1

	$removeByIndex index:
		if (index >= 0 && index < $__size)
			if (index == 0) $__root = $__root.next
			else
				itrtrNd = $__root
				i = 0
				while(i < index - 1)
					itrtrNd = itrtrNd.next
					i = i + 1
				itrtrNd.next = itrtrNd.next.next
			$__size = $__size - 1

	$get index:
		itrtrNd = $__root
		i = 0
		while(i < index)
			itrtrNd = itrtrNd.next
			i = i + 1
		itrtrNd.value


list = List.new()

println(list.size())

list.add(10)
list.add(20)

println(list.size())
println(list.get(0))
println(list.get(1))

list.add(30)
list.removeByIndex(1)
println(list.get(1))

list.removeByIndex(0)
println(list.get(0))

println(list.size())
