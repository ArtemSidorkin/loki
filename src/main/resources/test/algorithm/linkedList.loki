List:

	Node value next:
		$value = value
		$next = next

	$__root = none
	$__size = 0

	$size: $__size

	$add value:
		nwNd = Node.new(value, none)
		if ($__root == none) $__root = nwNd
		else
			itrtrNd = $__root
			while(itrtrNd.next != none) itrtrNd = itrtrNd.next
			itrtrNd.next = nwNd
		$__size = $__size + 1

	$removeByIndex index:
		if (index >= 0 && index < $__size)
			if (index == 0) $__root = $__root.next
			else
				itrtrNd = $__root
				i = 0
				while(i < index - 1)
					itrtrNd = itrtrNd.next
					i = i + 1
				itrtrNd.next = itrtrNd.next.next
			$__size = $__size - 1

	$get index:
		itrtrNd = $__root
		i = 0
		while(i < index)
			itrtrNd = itrtrNd.next
			i = i + 1
		itrtrNd.value


list = List.new()

assert(list.size(), 0)

list.add(10)
list.add(20)

assert(list.size(), 2)
assert(list.get(0), 10)
assert(list.get(1), 20)

list.add(30)
list.removeByIndex(1)
assert(list.get(1), 30)

list.removeByIndex(0)
assert(list.get(0), 30)

assert(list.size(), 1)
