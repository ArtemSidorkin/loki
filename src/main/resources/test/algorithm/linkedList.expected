0
2
10
20
30
30
1